src/mo_frontend/typing.ml

```diff
@@ -1419,7 +1460,7 @@ and check_exp' env0 t exp : T.typ =
   | TupE exps, T.Tup ts when List.length exps = List.length ts ->
     List.iter2 (check_exp env) ts exps;
     t
-  | ObjE exp_fields, T.Obj(T.Object, fts) ->
+  | ObjE ([], exp_fields), T.Obj(T.Object, fts) -> (* FIXME: bases? *)
```

Comment:
I guess you could do better here by inferring the non-empty bases but pushing the expected fields types into the fields, but that might duplicate a lot of code from the synthesis case above....

Reply:
Yeah, I guess if I had a test case that potentially gave a better error message then I'd be inclined more going down this path, but right now it looks like a lot of sweat for little gain...

Reply:
```suggestion
  | ObjE ([], exp_fields), T.Obj(T.Object, fts) -> (* TODO: infer bases? *)
```